src/js/atoms/modal.js

```diff
@@ -0,0 +1,92 @@
+// Transforms a normal YouTube URL into an iframe embed URL
+function transformYouTubeURL(url) {
+  const re = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)([^#&?]*).*/;
```

Comment:
Does this regex catch video URLs that aren't on YT's "cute" domain?

Reply:
It does because the `(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)` capture group is testing for strings that will directly precede the ID, but that does highlight a big problem with this regex – it’s way too permissive in what it accepts.

I’ve made some modifications to it:

```regex
/^https?:\/\/(?:(?:www\.)?youtube(?:-nocookie)?\.com.*(?:\/|v\/|u\/\w\/|embed\/|\?v=|&v=)|youtu.be\/)([\w-]{11})/
```

- String must start http:// or https://.
- Strict domain matching. Either youtube.com, youtube-nocookie.com – with or without www subdomain – or youtu.be.
- Explicit characters (`[\w-]`) and length `{11}` for the ID.
- Non-capturing groups throughout except the ID.

I’ve tried with a sample of YouTube IDs but will test against more before I commit it.